lib/net.js

```diff
@@ -524,16 +524,8 @@ Socket.prototype._read = function(n) {
 
 Socket.prototype.end = function(data, encoding) {
   stream.Duplex.prototype.end.call(this, data, encoding);
-  this.writable = false;
   DTRACE_NET_STREAM_END(this);
   LTTNG_NET_STREAM_END(this);
```

Comment:
Can we move these to `_destroy` instead and use a stock `end` implementation?

Reply:
I think would be a breaking change in light of your comment about not returning `this` below :/ I’d agree that that would be ideal, definitely.